cmd/provenanced/config/client.go

```diff
@@ -50,48 +49,48 @@ func (c *ClientConfig) SetBroadcastMode(broadcastMode string) {
 	c.BroadcastMode = broadcastMode
 }
 
-// ReadFromClientConfig reads values from client.toml file and updates them in client Context
-func ReadFromClientConfig(ctx client.Context) (client.Context, error) {
-	configPath := filepath.Join(ctx.HomeDir, "config")
-	configFilePath := filepath.Join(configPath, "client.toml")
-	conf := defaultClientConfig()
-
-	// if config.toml file does not exist we create it and write default ClientConfig values into it.
-	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
-		if err := ensureConfigPath(configPath); err != nil {
-			return ctx, fmt.Errorf("couldn't make client config: %v", err)
-		}
-
-		if err := WriteConfigToFile(configFilePath, conf); err != nil {
-			return ctx, fmt.Errorf("could not write client config to the file: %v", err)
-		}
+func (c ClientConfig) ValidateBasic() error {
+	switch c.KeyringBackend {
+	case "os", "file", "kwallet", "pass", "test", "memory":
+	default:
+		return errors.New("unknown keyring-backend (must be one of 'os', 'file', 'kwallet', 'pass', 'test', 'memory')")
 	}
-
-	conf, err := GetClientConfig(configPath, ctx.Viper)
-	if err != nil {
-		return ctx, fmt.Errorf("couldn't get client config: %v", err)
+	switch c.Output {
+	case "text", "json":
+	default:
+		return errors.New("unknown output (must be 'text' or 'json')")
+	}
+	switch c.BroadcastMode {
+	case "sync", "async", "block":
+	default:
+		return errors.New("unknown broadcast-mode (must be one of 'sync' 'async' 'block')")
 	}
+	return nil
+}
+
+// ApplyClientConfigToContext returns a client context after having the client config values applied to it.
+func ApplyClientConfigToContext(ctx client.Context, config *ClientConfig) (client.Context, error) {
 	// we need to update KeyringDir field on Client Context first cause it is used in NewKeyringFromBackend
-	ctx = ctx.WithOutputFormat(conf.Output).
-		WithChainID(conf.ChainID).
+	ctx = ctx.WithOutputFormat(config.Output).
+		WithChainID(config.ChainID).
 		WithKeyringDir(ctx.HomeDir)
 
-	keyring, err := client.NewKeyringFromBackend(ctx, conf.KeyringBackend)
-	if err != nil {
-		return ctx, fmt.Errorf("couldn't get key ring: %v", err)
+	keyring, krerr := client.NewKeyringFromBackend(ctx, config.KeyringBackend)
+	if krerr != nil {
+		return ctx, fmt.Errorf("couldn't get key ring: %v", krerr)
 	}
 
```

Comment:
We talked about making the error message a little clearer here.  Currently, the error message is `Error: could not apply client config to client context: couldn't get key ring: Specified keyring backend not available` and maybe we could add a little more detail  like you had suggested: `The keyring-backend value needs to be manually updated in <file>.` ?  Also, the current error doesn't state the name of the keyring entered `Specified keyring backend`, could we also add the `config.KeyringBackend` value?

Reply:
I can add some direction about updating it. Providing the location of where to make the change is difficult to iron out, though, because there's no way to tell if it was from a config file or an env var. But I guess I can provide the file and also mention env vars.

Working on the update now.

Reply:
Updated it to now look like this:
`Error: could not apply client config build/run/provenanced/config/client.toml to client context - it may need to be updated manually: couldn't get key ring kwallet: Specified keyring backend not available`

Reply:
That looks good to me!